### PR TITLE
sink: Encoder configuration unit tests

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- Lack of unit tests for the open protocol parameters `max-message-bytes` and `max-batch-size`.

### What is changed and how it works?
- Added unit tests for `max-message-bytes` and `max-batch-size`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has interface methods change

Related changes

 - Need to cherry-pick to the release branch

### Release note

- No release note

### CODE REVIEW

File: cdc/sink/mq_test.go [hunk 1]
```diff
@@ -182,3 +190,30 @@ func (s mqSinkSuite) TestKafkaSinkFilter(c *check.C) {
...
+func (s mqSinkSuite) TestPulsarSinkEncoderConfig(c *check.C) {
+	defer testleak.AfterTest(c)()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := failpoint.Enable("github.com/pingcap/ticdc/cdc/sink/producer/pulsar/MockPulsar", "return(true)")
+	c.Assert(err, check.IsNil)
+
+	uri := "pulsar://127.0.0.1:1234/kafka-test?max-batch-size=1" +
+		"&max-message-bytes=4194304&max-batch-size=1048576"
```
Comment (amyangfei) [inline cdc/sink/mq_test.go:203]: Why there are two `max-batch-size`

Comment (overvenus) [inline cdc/sink/mq_test.go:203]: Please test common edge cases, -1, 0, i32max, u32max, etc.

Comment (liuzix) [reply cdc/sink/mq_test.go:203]: It was in the original Kafka test case. Fixed.

File: cdc/sink/codec/json.go [hunk 2]
```diff
@@ -519,21 +531,27 @@ func (d *JSONEventBatchEncoder) SetParams(params map[string]string) error {
...
-			return errors.Trace(err)
+			return cerror.ErrKafkaInvalidConfig.Wrap(err)
 		}
 	} else {
-		d.maxBatchSize = 4096
+		d.maxBatchSize = DefaultMaxBatchSize
+	}
+
+	if d.maxBatchSize <= 0 {
+		return cerror.ErrKafkaInvalidConfig.Wrap(errors.Errorf("invalid max-batch-size %d", d.maxKafkaMessageSize))
```
Comment (amyangfei) [inline cdc/sink/codec/json.go:554]:
```suggestion
		return cerror.ErrKafkaInvalidConfig.Wrap(errors.Errorf("invalid max-batch-size %d", d.maxBatchSize))
```

**Review by amyangfei — COMMENTED** [review]
LGTM

**Review by leoppro — APPROVED** [review]

**Review by ti-srebot — APPROVED** [review]
